### PR TITLE
Create deployment command

### PR DESCRIPTION
**Pull request by andrea-mosk**
### What and why?

Create a `datadog-ci deployment mark` command that can be used to mark a CI Job as a deployment and be shown in CD Visibility. 

### How?

Behind the scenes, this command uses the `datadog-ci tag command` setting specific deployment-related tags that are processed in a different way.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

**Review by romainkomorndatadog — APPROVED** [review]

**Review by juan-fernandez — APPROVED** [review]

File: src/commands/deployment/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Deployment command
+
+Marks a CI job as a deployment. This command uses the datadog-ci tag command by generating specific tags that are
```
[the remark this answers is not in the document]
Comment (andrea-mosk) [reply src/commands/deployment/README.md]: Makes sense, removed

File: src/commands/deployment/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Marks a CI job as a deployment.
+
+## Usage
+
+### Commands
+
+#### `mark`
+
+The mark command specifies that a CI Job executes a deployment.
```
Comment (drichards-87) [inline src/commands/deployment/README.md:11]:
```suggestion
The `mark` command specifies that a CI job executes a deployment.
```

File: src/commands/deployment/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+- `--env` is the environment to which this deployment is performed. For example, `prod`.
+- `--revision` is the revision/version that is being deployed. For example, `1.0.0` or `v123-456`.
+- `--is-rollback` specifies that the deployment is a rollback.
+- `--tags` is an array of key value pairs of the shape `key:value`. These tags will be added to the deployment event shown in Datadog.
+- `--no-fail` (default: `false`) will prevent the deployment command from failing if there are issues submitting the data.
+
+### Environment variables
+
+- `DD_API_KEY` (**required**): API key used to authenticate the requests.
+- `DD_SITE`: choose your Datadog site, e.g. datadoghq.com or datadoghq.eu.
```
Comment (drichards-87) [inline src/commands/deployment/README.md:32]:
```suggestion
- `DD_SITE`: choose your Datadog site. For example, datadoghq.com or datadoghq.eu.
```

File: src/commands/deployment/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+For example:
+
+```bash
+datadog-ci deployment mark --env prod --revision v1.1.0 --tags team:backend --no-fail
+```
+
+- `--env` is the environment to which this deployment is performed. For example, `prod`.
+- `--revision` is the revision/version that is being deployed. For example, `1.0.0` or `v123-456`.
+- `--is-rollback` specifies that the deployment is a rollback.
+- `--tags` is an array of key value pairs of the shape `key:value`. These tags will be added to the deployment event shown in Datadog.
```
Comment (drichards-87) [inline src/commands/deployment/README.md:26]:
```suggestion
- `--tags` is an array of key value pairs in the format `key:value`. These tags are added to the deployment event shown in Datadog.
```

**Review by drichards-87 — APPROVED** [review]
Left some minor suggestions from Docs and approved the PR.

**Review by Drarig29 — APPROVED** [review]
LGTM

File: src/commands/deployment/mark.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import {Command, Option} from 'clipanion'
+
+import {TagCommand} from '../tag/tag'
+
+import {CUSTOM_TAGS_TAG, ENV_TAG, IS_DEPLOYMENT_TAG, IS_ROLLBACK_TAG, REVISION_TAG} from './constants'
+
+/**
+ * This command is a wrapper around the datadog-ci tag command, allowing customers to mark CI Jobs
```
Comment (Drarig29) [inline src/commands/deployment/mark.ts:8]: nit: (to align with https://github.com/DataDog/datadog-ci/pull/1141#discussion_r1425524579)
```suggestion
 * This command is a wrapper around the datadog-ci tag command, allowing customers to mark CI jobs
```

**Pull request merged**